src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
Re:
1- Yes, memory is an issue
2- I'm just talking about a "virtual/logical" partitioning in the inner working of the comparator to remove the "last" one, in no way a hard partition that prevents an unified storage that can share the information to service requests from multiple DHTs.
3- You are right (this a good point)

The second paragraph is: Why your idea is better? and why the need to move the logic outside of the `dht_default_storage`?